addon/instance-initializers/user-agent.js

```diff
@@ -0,0 +1,27 @@
+/* global FastBoot */
+
+import { get, set } from '@ember/object';
+import { assert } from '@ember/debug';
+
+function getUserAgent(appInstance) {
+  if (typeof FastBoot === 'undefined') {
```

Comment:
Having this Fastboot-specific code in a separate file would allow us to filter it out at build time.

Something like: https://github.com/ronco/fastboot-filter-initializers